### ISSUE TITLE
2.7 RC-1: Single select autocomplete fields have a default value

### ISSUE THREAD

**Issue by lougreenwood** [pods-framework/pods#4457]
Using 2.7 RC-1, when creating a new record in WP-Admin for a Pod which has a single select autocomplete relationship field, the field will have a default value pre-set.

It seems that the default value set for the field is the first related entity by alphabetical order.

<img width="949" alt="screen shot 2017-09-08 at 12 24 23" src="https://user-images.githubusercontent.com/12051736/30215171-f43a54e4-94a6-11e7-9561-c4f59c5c64a3.png">

**Comment by sc0ttkclark** [comment]
@pglewis for your hit list

**Comment by pglewis** [comment]
This doesn't appear to be a data issue... the data correctly has all items set as `selected: false` and switching to drop-down as the view works correctly.  So it's probably something in the select2 initialization.

**Comment by jamesgol** [comment]
First off, a workaround is to set the field as multiple but with a limit of 1.  Though it does change the appearance when you have that item selected.

The problem here is that select2 handles multi vs single differently, trying to keep it as a direct replacement for <select>.  So to make it work (and I have done a POC but it's a hack since I'm not fully versed in the dfv) you need to have the first element have an id of something like '-1' and be empty.

My hack POC was adding
```        
array_unshift( $data['fieldItemData'], array( 'id' => '-1', 'selected' => false ) );
```
to PodsField->build_dfv_field_data();

then in ui/js/pods-dfv/_src/pick/views/select-view.js->setupSelect2 adjust the select2Options to:
```
		select2Options = {
			maximumSelectionLength: limit,
			placeholder           : { id : '-1', text: placeholder },
			allowClear            : ( 'single' === fieldConfig.pick_format_type ),
			disabled              : fieldConfig.limitDisable,
			escapeMarkup          : function ( text ) { return text; }
		};
```

This did uncover another bug I'll open an issue on.

**Comment by pglewis** [comment]
Aha, we probably just need to include autocomplete as well as dropdown here: 

https://github.com/pods-framework/pods/blob/23e0656/classes/fields/pick.php#L1644

**Comment by pglewis** [comment]
I finally have a magic incantation that should fix this, PR: #4558

Two sticking points that made this take longer to figure out than it should have: 

1) For some reason I had to specify `selected='selected'` for the empty placeholder option though none of the examples nor a simple prototype needed this.  

2) `maximumSelectionLength` can't be set to one for this specific case and should probably remain unset for single select, leaving the limit enforcement to select2

**Comment by pglewis** [comment]
Ready for you to test @lougreenwood, this is one of the final ones for 2.7
